added functions

SwagataBhattacharyya opened the pull request:
add verification functions

SwagataBhattacharyya committed:
added functions

SwagataBhattacharyya committed:
added functions

SwagataBhattacharyya committed:
added functions

SwagataBhattacharyya committed:
updated function description

SwagataBhattacharyya committed:
updated functions

SwagataBhattacharyya committed:
updated functions

SwagataBhattacharyya committed:
Updated

SwagataBhattacharyya committed:
removed utils

SwagataBhattacharyya commented:
f136895 added functions 1. verify_avaln and verify_alloq 2. changes to pk_variables: For derived variable AVALN=0, added BLQ, BQL, BLLOQ, BLOQ to grepl patterns (<BQL ,<BLOQ) 
cba47f4 resolved merge issues 
0ba5363 updated function @description, @param, @return for verify_avaln and verify_alloq
75ec841 1. changes to check_time_variables to substitute code with an equivalent for min_no_na from bsd.report 2. changes to match_missing_values to address warning related to funs()
e34dc21 changes in spacing of brackets in check_time_variables
71ffeba removed importFrom(utils,View) and importFrom(utils,read.csv) from namespace

SwagataBhattacharyya committed:
changes to check_units.R to address warning related to funs()

SwagataBhattacharyya commented:
@billdenney this branch is ready for review

billdenney commented on R/check_time_variables.R:
```diff
@@ -91,7 +92,8 @@ check_time_variables<-function(data){
       is.na(TSFM) & is.na(TAD)~1,
-      suppressWarnings(min(TSFM[EVID %in% 1]) %in% NA_integer_) & is.na(TAD) ~1,
+      suppressWarnings(min(TSFM[EVID %in% 1]) %in% NA_integer_ & is.na(TAD)) ~1,
+      suppressWarnings(min(TSFM[EVID %in% 1]) %in% "Inf" & is.na(TAD))~1,
```
Why are there warnings for these two lines?  Can we remove suppressWarnings() or otherwise prevent the warnings?  (I generally worry that `suppressWarnings()` may hide something that we want to see.)

SwagataBhattacharyya replied:
Below warning message appears when attempting to find the minimum value of a vector of length zero. eg. min(TSFM[EVID %in% 1]) results in numeric(0):
_In min(data) : no non-missing arguments to min; returning Inf._

Created function min_no_warning() to address the above warning issue and added code lines for vectors with min values in NA or Inf, removed suppressWarnings(): 
   **is.na**(min_no_warning(var1[EVID` %in% 1])) & is.na(var2) ~1
   min_no_warning(var1[EVID %in% 1]) %in% **Inf** & is.na(var2)~1

05c7f00 Resolved issue related to warning for check_time_variables.R

Added variables STUDYID,USUBJID,EVID to check_var

billdenney replied:
@SBcytel Thanks!  That's exactly the way that I was hoping a fix would be made (though I didn't know the warning that was being covered).  Now, we see warnings if there is a reason we should, but we don't worry about the zero-length case.

billdenney reviewed (commented):
I only have the one question/comment about suppressWarnings.  With that addressed, we can merge.

SwagataBhattacharyya committed:
check_time_variables.R
Below warning message appears when attempting to find the minimum value of a vector of length zero. Eg. min(TSFM[EVID %in% 1]) results in numeric(0):
In min(data) : no non-missing arguments to min; returning Inf.
Created function min_no_warning() to address the above warning issue.
Added code lines for vectors with min values in NA or Inf:
   is.na(min_no_warning(var1[EVID %in% 1])) & is.na(var2) ~1
   min_no_warning(var1[EVID %in% 1]) %in% Inf & is.na(var2)~1

pk_variables.R
Added ignore_case=TRUE to grepl
